framework/src/actions/SetupDebugAction.C

```diff
@@ -65,6 +66,11 @@ SetupDebugAction::act()
     MooseObjectAction * action = createOutputAction("TopResidualDebugOutput", "_moose_top_residual_debug_output");
     action->getObjectParams().set<unsigned int>("num_residuals") = _pars.get<unsigned int>("show_top_residuals");
   }
+
+  // Material dependency
+  if (_pars.get<bool>("show_material_dependency"))
+    for (THREAD_ID tid = 0; tid < libMesh::n_threads(); ++tid)
+      _problem->getMaterialWarehouse(tid).setDebugDependency(true);
```

Comment:
This seems a little odd, Are we really going to print information in a threaded region?